### PR TITLE
Adding Travis configuration

### PR DESCRIPTION
**Pull request by buinauskas**
@Razzeee Hey, I've picked up #36 and have made the following changes:
-   Added Travis configuration
-   Configured pre-commit hook to automatically format staged code
-   Configured pre-commit hook to automatically lint files
-   Fixed linting issues project had (very minor ones)
-   Added CI build step to check if code is actually formatted as per spec
-   Changed few typos (Will submit another PR to fix more of these)

I've never worked with Travis CI before, so apologies if I did something incorrectly - I'd be glad if I can make this work. I've tested this on my fork and build was successful. Let me know if there's something I can improve, thanks!

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+language: node_js
+
+env:
+  - CXX=clang++
+
+branches:
+  only:
+    - master
```
Comment (Razzeee) [inline .travis.yml]: Hrm, I think we should build on all branches. That way we can see if something is wrong, before it get's merged :)

Comment (buinauskas) [reply .travis.yml]: Sure, totally makes sense :+1:  Removed restriction.

File: package.json [hunk 2]
```diff
@@ -42,9 +45,16 @@
   "scripts": {
     "compile": "tsc -b",
     "watch": "tsc -b -w",
+    "format:check": "prettier --config ./.prettierrc.yaml --list-different \"src/**/*{.ts,.js,.json}\"",
```
Comment (Razzeee) [inline package.json]: Do we need this? Shouldn't "lint" report these too, as we're using tslint-plugin-prettier?

**Comment by Razzeee** [comment]
Thanks for taking the time to do this. I just found some minors, after that  it's ready to go in.

File: package.json (hunk 2 above)
Comment (buinauskas) [reply package.json]: I've always used this kind of approach, but I'm happy to check if tslint-plugin-prettier does exactly the same. Will update based on my findings :)

Comment (buinauskas) [reply package.json]: Reverted.

Comment (buinauskas) [reply package.json]: Removed redundant check :+1:

File: .travis.yml (hunk 1 above)
Comment (buinauskas) [reply .travis.yml]: This is rather silly, but should I resolve this conversation upon resolving issues or it's better if you'd check that and close it?

Comment (Razzeee) [reply .travis.yml]: I like to have others check and resolve these discussions :)

Comment (Razzeee) [reply .travis.yml]: So in this case me ;)

**Pull request merged**